tests/test_examples.py

```diff
@@ -0,0 +1,78 @@
+import itertools
+import sys
+from pathlib import Path
+from subprocess import PIPE  # noqa: S404
+from subprocess import Popen  # noqa: S404
+from subprocess import STDOUT  # noqa: S404
+from time import sleep
+from typing import List
+
+import pytest
+
+HERE = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve().parent
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def path_1_simple() -> Path:
+    path = (HERE / ".." / "examples" / "1_simple").resolve()
+    assert path.exists()
+    return path
+
+
+@pytest.fixture
+def path_2_base_time_add() -> Path:
+    path = (HERE / ".." / "examples" / "2_base_time_add").resolve()
+    assert path.exists()
+    return path
+
+
+# UTILS
+
+
+def assert_run_in_parallel(scrips_to_run: List[Path]) -> None:
+    # check all exist before running
+    for script_to_run in scrips_to_run:
+        assert script_to_run.exists()
+
+    for permutation in itertools.permutations(scrips_to_run):
+        # run provided scripts
+        print(f"Running permutation {permutation}")
+        processes = [
+            Popen(  # noqa: S607
```

Comment:
Why dont' you use ``subprocess.run`` instead? the latest versions of python provide this more convenient helper that is far more readable and has preconfigured the most common setups. Using raw Popen can be sometimes errorprone ...  (we already had this discussion several times... :-) )

Reply:
Not what I wanted or needed. From [the docs](https://docs.python.org/3/library/subprocess.html#subprocess.run)

> Run the command described by args. **Wait for command to complete**, then return a [CompletedProcess](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess) instance.

I need to start 2 scripts and wait for them to complete.